base/socket.jl

```diff
@@ -2,6 +2,7 @@
 abstract IPAddr
 
 Base.isless{T<:IPAddr}(a::T, b::T) = isless(a.host, b.host)
+Base.convert{T<:Integer}(dt::Type{T}, ip::IPAddr) = dt(ip.host)
```

Comment:
@pao that's what I was originally thinking, but now I guess I've confused myself. How then is `T(foo)` the same as `dt(foo)` in the original code?